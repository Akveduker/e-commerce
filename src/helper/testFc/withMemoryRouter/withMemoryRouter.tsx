import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"

export const withMemoryRouter = (children: React.ReactNode) => {
    return render(
        <MemoryRouter>
            {children}
        </MemoryRouter>
    )
}