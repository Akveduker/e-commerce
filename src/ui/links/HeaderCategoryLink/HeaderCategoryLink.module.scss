@import "../../../styles/_mixins.scss";
.link {
    @include fontMedium;
    font-size: 15px;
    display: flex;
    align-items: center;
}
