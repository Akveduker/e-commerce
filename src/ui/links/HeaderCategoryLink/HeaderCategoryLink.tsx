import React, { FC } from 'react';
import { Link, LinkProps } from 'react-router-dom';
import s from './HeaderCategoryLink.module.scss'
interface HeaderCategoryLinkProps extends LinkProps {
    children: React.ReactNode;
}
const HeaderCategoryLink: FC<HeaderCategoryLinkProps> = ({ children, ...props }) => {
    return (
        <Link {...props} className={s['link']}>
            {children}
        </Link>
    );
};

export default HeaderCategoryLink;