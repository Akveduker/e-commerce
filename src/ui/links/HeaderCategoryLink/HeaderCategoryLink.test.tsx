import { render } from "@testing-library/react"
import { withMemoryRouter } from "../../../helper/testFc/withMemoryRouter/withMemoryRouter"
import HeaderCategoryLink from "./HeaderCategoryLink"

describe('HeaderCategoryLink', () => {
    test('Снапшот', () => {
        const { container } = withMemoryRouter(
            <HeaderCategoryLink to={'/'}>
                link
            </HeaderCategoryLink>
        )
        expect(container).toMatchSnapshot()
    })
})