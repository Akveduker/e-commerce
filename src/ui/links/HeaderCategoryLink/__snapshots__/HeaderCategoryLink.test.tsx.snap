// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderCategoryLink Снапшот 1`] = `
<div>
  <a
    class="link"
    href="/"
  >
    link
  </a>
</div>
`;
